### ISSUE TITLE
Pre-compute mask for N2V during validation

### ISSUE THREAD

**Issue by veegalinova** [CAREamics/careamics-restoration#703]
## Description

Raised in [#701](https://github.com/CAREamics/careamics/pull/701#discussion_r2732564264)
N2V pixel masking is currently applied on-the-fly during both training and validation steps. Before, the mask was pre-computed for the validation step ([see here](https://github.com/juglab/n2v/blob/c61b3b47c5cacc533f7e0e3db7cefab139d7b62a/n2v/models/n2v_standard.py#L247)). 
Currently, this makes validation loss not comparable between epochs, unless the same seed is used for all validation masking operations.
